src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        arrayCapacityCheck();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        size++;
+        indexCheck(index);
+        arrayCapacityCheck();
+        System.arraycopy(elementData, index, elementData,
+                (index + 1), elementData.length - index - 1);
+        elementData[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size();i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheck(index);
+        return elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexCheck(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheck(index);
+        T temp = elementData[index];
+        System.arraycopy(elementData,(index + 1),
+                elementData, index, (elementData.length - index - 1));
+        size--;
+        return temp;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        T temp = null;
+        int count = 0;
+        for (int i = 0; i < size;i++) {
+            if ((element == null && elementData[i] == null)
+                    || (elementData[i] != null && (elementData[i].equals(element)))) {
+                temp = elementData[i];
+                remove(i);
+                count++;
+                break;
+            }
+        }
+        noSuchElementCheck(count);
+        return temp;
```

Comment:
we don't need this `count` and `temp` vars and `noSuchElementCheck()` method, we can just return the values right in the loop if we found it, and throw an exception after the loop if we did not 
+ `remove(index)` gets you the element, so you can use it
+ we can simplify if conditon
```suggestion
        for (int i = 0; i < size;i++) {
            if ((element == elementData[i] || element != null && element.equals(elementData[i]) {
                return remove(i);
            }
        }
        throw new NoSuchElementException(...);
```